main/src/cgeo/geocaching/connector/su/SuConnector.java

```diff
@@ -0,0 +1,249 @@
+
+    private static final CharSequence PREFIX_MULTISTEP_VIRTUAL = "MV";
+    private static final CharSequence PREFIX_TRADITIONAL = "TR";
+    private static final CharSequence PREFIX_VIRTUAL = "VI";
+    private static final CharSequence PREFIX_MULTISTEP = "MS";
+    private static final CharSequence PREFIX_EVENT = "EV";
+    private static final CharSequence PREFIX_CONTEST = "CT";
+    private static final CharSequence PREFIX_MYSTERY = "LT";
+    private static final CharSequence PREFIX_MYSTERY_VIRTUAL = "LV";
+
+    // Let just add this general prefix, since for search prefix is not important at all,
+    // all IDs are unique at SU
+    private static final CharSequence PREFIX_GENERAL = "SU";
+
+
+    @NonNull
+    private final SuLogin suLogin = SuLogin.getInstance();
+
+    private SuConnector() {
+        // singleton
+    }
+
+    public static SuConnector getInstance() {
+        return Holder.INSTANCE;
+    }
+
+    @Override
+    public boolean login(final Handler handler, @Nullable final Activity fromActivity) {
+        final StatusCode status = suLogin.login();
+
+        if (ConnectorFactory.showLoginToast && handler != null) {
+            handler.sendMessage(handler.obtainMessage(0, status));
+            ConnectorFactory.showLoginToast = false;
+
+            // invoke settings activity to insert login details
+            if (status == StatusCode.NO_LOGIN_INFO_STORED && fromActivity != null) {
+                SettingsActivity.openForScreen(R.string.preference_screen_su, fromActivity);
+            }
+        }
+        return status == StatusCode.NO_ERROR;
+    }
+
+    @Override
+    public String getUserName() {
+        return suLogin.getActualUserName();
+    }
+
+    @Override
+    public int getCachesFound() {
+        return suLogin.getActualCachesFound();
+    }
+
+    @Override
+    public String getLoginStatusString() {
+        return suLogin.getActualStatus();
+    }
+
+    @Override
+    public boolean isLoggedIn() {
+        return suLogin.isActualLoginStatus();
+    }
+
+    @Override
+    public Credentials getCredentials() {
+        return Settings.getCredentials(R.string.pref_suusername, R.string.pref_supassword);
+    }
+
+    @Override
+    public int getUsernamePreferenceKey() {
+        return R.string.pref_suusername;
+    }
+
+    @Override
+    public int getPasswordPreferenceKey() {
+        return R.string.pref_supassword;
+    }
+
+    @Override
+    public int getAvatarPreferenceKey() {
+        return R.string.pref_su_avatar;
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return "Geocaching.su";
+    }
+
+    @Override
+    @NonNull
+    public String getNameAbbreviated() {
+        return "GCSU";
+    }
+
+    @Override
+    @NonNull
+    public String getCacheUrl(@NonNull final Geocache cache) {
+        return getCacheUrlPrefix() + "&cid=" + cache.getCacheId();
+    }
+
+    @Override
+    public boolean getHttps() {
+        return false;
```

Comment:
No chance for https? I mean, setting up let's encrypt is supposed to be quite easy.